src/OpenSolid/BoundingBox2d.elm

```diff
@@ -342,27 +361,271 @@ contains point boundingBox =
     (minX boundingBox <= maxX other)
         && (maxX boundingBox >= minX other)
         && (minY boundingBox <= maxY other)
         && (maxY boundingBox >= minY other)
 
 
-strictlyOverlaps : BoundingBox2d -> BoundingBox2d -> Bool
-strictlyOverlaps other boundingBox =
-    (minX boundingBox < maxX other)
-        && (maxX boundingBox > minX other)
-        && (minY boundingBox < maxY other)
-        && (maxY boundingBox > minY other)
+{-| DEPRECATED: Alias for `intersects`, kept for compatibility. Use `intersects`
+instead.
+-}
+overlaps : BoundingBox2d -> BoundingBox2d -> Bool
+overlaps =
+    intersects
+
+
+overlapAmount : BoundingBox2d -> BoundingBox2d -> Maybe Float
+overlapAmount firstBox secondBox =
+    let
+        xOverlap =
+            min (maxX firstBox) (maxX secondBox)
+                - max (minX firstBox) (minX secondBox)
+
+        yOverlap =
+            min (maxY firstBox) (maxY secondBox)
+                - max (minY firstBox) (minY secondBox)
+    in
+    if xOverlap >= 0 && yOverlap >= 0 then
+        Just (min xOverlap yOverlap)
+    else
+        Nothing
+
+
+squaredSeparationAmount : BoundingBox2d -> BoundingBox2d -> Maybe Float
+squaredSeparationAmount firstBox secondBox =
+    let
+        xSeparation =
+            max (minX firstBox) (minX secondBox)
+                - min (maxX firstBox) (maxX secondBox)
+
+        ySeparation =
+            max (minY firstBox) (minY secondBox)
+                - min (maxY firstBox) (maxY secondBox)
+    in
+    if xSeparation > 0 && ySeparation > 0 then
+        Just (xSeparation * xSeparation + ySeparation * ySeparation)
+    else if xSeparation > 0 then
+        Just (xSeparation * xSeparation)
+    else if ySeparation > 0 then
+        Just (ySeparation * ySeparation)
+    else if xSeparation == 0 || ySeparation == 0 then
+        Just 0
+    else
+        Nothing
+
+
+alwaysFalse : BoundingBox2d -> BoundingBox2d -> Bool
+alwaysFalse firstBox secondBox =
+    False
+
+
+{-| Check if one box overlaps another by less than, greater than or equal to a
+given amount. You could perform a tolerant collision check (one that only
+returns true only if the boxes overlap by at least some small finite amount,
+and ignores boxes that just barely touch each other) as
+
+    boxesCollide =
+        BoundingBox2d.overlapsBy GT 0.001 firstBox secondBox
+
+(The [`Order`](http://package.elm-lang.org/packages/elm-lang/core/latest/Basics#Order)
+type and its three values `LT`, `GT` and `EQ` are defined in Elm's `Basics`
+module so are available by default in any Elm program.)
+
+Overlap is defined as the _minimum_ distance one box would have to move so that
+it did not touch the other, and is always positive for any two overlapping
+boxes.
+
+Boxes that just touch are considered to have an overlap of zero, which is
+distinct from 'no overlap'. Boxes that do not touch or overlap at all are
+considered to have an overlap which is less than zero but not greater or less
+than any negative number.
```

Comment:
This left me wondering what it meant that

>Boxes that do not touch or overlap at all are considered to have an overlap which is less than zero but **not greater or less than any negative number**.

I don’t understand why “is less than zero” wouldn’t be an exhaustive description.

Reply:
Oh, just after posting I read that sentence once more yet and it struck me what it means!

Perhaps we could add an encouraging phrase at the end of that paragraph, saying:

> Boxes that do not touch or overlap at all are considered to have an overlap which is less than zero but not greater or less than any negative number. **If you’re wondering what this means, the examples below will hopefully clear things up.**

Reply:
Yeah, something like what you suggests sounds great! I'd say go ahead and make whatever tweaks like that you think are necessary and I can do a final pass at the end.